File: trip-service-java/src/test/java/com/cawadev/softwarecraft/katas/tripservice/user/UserTest.java
package com.cawadev.softwarecraft.katas.tripservice.user;

import org.junit.jupiter.api.Test;

import static com.cawadev.softwarecraft.katas.tripservice.user.UserBuilder.builder;
import static org.assertj.core.api.Assertions.assertThat;

class UserTest {

    private static final User PAUL = new User();
    private static final User JACQUES = new User();

    @Test
    void should_return_true_when_user_is_friends_with_another_user() {
        var user = builder().friendsWith(PAUL).build();

        assertThat(user.isFriendsWith(PAUL)).isTrue();
        assertThat(user.isFriendsWith(JACQUES)).isFalse();
    }
}

File: trip-service-kotlin/src/test/kotlin/com/cawadev/softwarecraft/katas/tripservice/user/UserTest.kt
package com.cawadev.softwarecraft.katas.tripservice.user

import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

class UserTest {

    companion object {
        private val PAUL = User()
        private val TRISHA = User()
    }

    @Test
    fun `should return true when user are friends`() {
        val user = User().apply {
            addFriend(PAUL)
        }

        assertTrue(user.isFriendsWith(PAUL));
    }

    @Test
    fun `should return false when user are not friends`() {
        val user = User()

        assertFalse(user.isFriendsWith(TRISHA));
    }
}
